cmd/internal/find/primary/meta/objectPredicate.go

```diff
@@ -71,6 +53,51 @@ func parseObjectP(tokens []string, baseCaseParser, keySequenceParser predicate.P
 	return objectP(key, p), tokens, err
 }
 
+// A key consists of one or more characters that aren't
+// ".", "[", or "]". The reason we have this limitation
+// is so we can support key sequences, which includes
+// nested keys like ".key1.key2", ".key1[?].key2", etc.
+// key sequences aren't specified in the grammar because
+// they make it harder to formalize the semantics.
+//
+// NOTE: Users can still specify ".", "[", or "]" by escaping
+// them with a backslash "\". For example, '.com\.docker\.compose'
+// would be parsed as the key "com.docker.compose".
+func parseKeySequence(tk string) (string, string, error) {
+	isTerminatingChar := func(char byte) bool {
+		return char == '.' || char == '[' || char == ']'
+	}
+
+	if len(tk) <= 0 || tk[0] != '.' {
+		return "", "", errz.NewMatchError("key sequences must begin with a '.'")
+	}
+	key := ""
+	rem := tk[1:]
+	for {
+		if len(rem) <= 0 {
+			break
+		}
+		if isTerminatingChar(rem[0]) {
+			break
+		}
+		if rem[0] == '\\' {
+			// Lookahead and check for an escaped character
+			if len(rem) > 1 && isTerminatingChar(rem[1]) {
+				key += string(rem[1])
```

Comment:
These 3 lines could be `rem = rem[1:]` and still work. It says: this is a valid escape, remove the backslash and continue to add it to the key.

Reply:
Updated.